File: 25. Exam Preparation/Bunny-Wars - Author's solutions/C#-Solution/BunnyWars.Tests/Performance/AddRoomPerformance.cs
namespace BunnyWars.Tests.Performance
{
    using System;
    using System.Diagnostics;
    using BunnyWars.Core;
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class AddRoomPerformance : BaseTestClass
    {
        public Random Random { get; private set; }
        
        [TestInitialize]
        public override void Initialize()
        {
            base.Initialize();
            this.Random = new Random();
        }

        [TestCategory("Performance")]
        [TestMethod]
        [Timeout(250)]
        public void PerformanceAddRoom_With50000Rooms()
        {
            //Arrange
            var count = 1;
            var roomsCount = 50000;

            var sw = new Stopwatch();
            sw.Start();
            //Act
            for (int i = 0; i < roomsCount; i++)
            {
                this.BunnyWarCollection.AddRoom(i);
                Assert.AreEqual(count++,this.BunnyWarCollection.RoomCount,"Incorrect count of rooms after adding!");
            }
        }
    }
}

File: 25. Exam Preparation/Bunny-Wars - Author's solutions/Java-Skeleton-Eclipse/src/test/performance/AddRoomPerformance.java
package test.performance;

import org.junit.Assert;
import org.junit.Test;
import org.junit.experimental.categories.Category;
import test.helpers.BasePerformanceTest;
import test.types.PerformanceTests;

public class AddRoomPerformance extends BasePerformanceTest {
    @Category(PerformanceTests.class)
    @Test(timeout = 250)
    public void PerformanceAddRoom_With50000Rooms()
    {
        //Arrange
        int count = 1;
        int roomsCount = 50000;

        //Act
        for (int i = 0; i < roomsCount; i++)
        {
            this.BunnyWarCollection.addRoom(i);
            Assert.assertEquals("Incorrect count of rooms after adding!", count++, this.BunnyWarCollection.getRoomCount());
        }
    }

}
